app_dart/lib/src/service/scheduler.dart

```diff
@@ -200,20 +208,68 @@ class Scheduler {
       ));
     });
 
+    final SchedulerConfig schedulerConfig = await getSchedulerConfig(commit);
+    log.debug('Loaded scheduler config $schedulerConfig');
+
     return tasks;
   }
 
+  /// Load in memory the `.ci.yaml`.
+  Future<SchedulerConfig> getSchedulerConfig(Commit commit, {RetryOptions retryOptions}) async {
+    final String ciPath = '${commit.repository}/${commit.sha}/.ci.yaml';
+    final Uint8List configBytes = await cache.getOrCreate(subcacheName, ciPath,
+        createFn: () => _downloadSchedulerConfig(
+              ciPath,
+              retryOptions: retryOptions,
+            ),
+        ttl: const Duration(hours: 1));
+    return SchedulerConfig.fromBuffer(configBytes);
+  }
+
+  /// Get `.ci.yaml` from GitHub, and store the bytes in redis for future retrieval.
+  ///
+  /// If GitHub returns [HttpStatus.notFound], an empty config will be inserted assuming
+  /// that commit does not support the scheduler config file.
+  Future<Uint8List> _downloadSchedulerConfig(String ciPath, {RetryOptions retryOptions}) async {
+    String configContent;
+    try {
+      configContent = await githubFileContent(
+        ciPath,
+        httpClientProvider: httpClientProvider,
+        log: log,
+        retryOptions: retryOptions,
+      );
+    } on NotFoundException {
+      return SchedulerConfig.getDefault().writeToBuffer();
+    }
+    final YamlMap configYaml = loadYaml(configContent) as YamlMap;
+    return schedulerConfigFromYaml(configYaml).writeToBuffer();
+  }
+
   /// Load in memory the Cocoon Agent DeviceLab scheduler config.
   ///
+  /// If GitHub returns [HttpStatus.notFound], an empty manifest is returned.
+  ///
   // TODO(chillers): Remove when DeviceLab has migrated to LUCI. https://github.com/flutter/flutter/projects/151
   @visibleForTesting
-  Future<YamlMap> loadDevicelabManifest(Commit commit) async {
-    final String path = '/${commit.repository}/${commit.sha}/dev/devicelab/manifest.yaml';
+  Future<YamlMap> loadDevicelabManifest(
+    Commit commit, {
+    RetryOptions retryOptions,
+  }) async {
+    final String path = '${commit.repository}/${commit.sha}/dev/devicelab/manifest.yaml';
```

Comment:
Ditto. a `/` needed?

Reply:
this is actually the format that `githubFileContent` expects.

Reply:
I updated the format as if everything passes a leading slash, `githubFileContent` should just add it.